pkg/cmd/server/start/start_node.go

```diff
@@ -249,30 +273,39 @@ func (o NodeOptions) IsRunFromConfig() bool {
 	return (len(o.ConfigFile) > 0)
 }
 
-func StartNode(nodeConfig configapi.NodeConfig) error {
+func StartNode(nodeConfig configapi.NodeConfig, nodeArgs *NodeArgs) error {
 	config, err := kubernetes.BuildKubernetesNodeConfig(nodeConfig)
 	if err != nil {
 		return err
 	}
-	glog.Infof("Starting node %s (%s)", config.KubeletServer.HostnameOverride, version.Get().String())
+
+	if nodeArgs.Components.Enabled("kubelet") {
```

Comment:
constants for component names (used here and in the flag bindings and validation)